### PR TITLE
Jc/add school contacts to schools

### PR DESCRIPTION
**Pull request by JamieCleare2525**
## Context

- School users can create a SchoolContact for their organisation.

## Changes proposed in this pull request

- Controllers/models/routes/etc. Everything you need to create a SchoolContact belonging to a School.

## Guidance to review

- Sign in as Anne
- Click "Organisation details" in the top nav
- Scroll to the bottom of the page.
- There should be a `div` to allow users to add a School Contact
- Click "Add school contact"
- Enter a name and email address (to test validation, enter an email address which is not an email address format)
- Click "Continue"
- You should be directed to a "Check your answers page" (feel free to test the change and back links)
- Click "Add school contact"
- You should have created a School Contact for your organisation

- Test editing a School Contact, by using the change links next to each attribute of the School Contact.
- When updating the attributes you will not be redirected to a check your answers.

- Test removing the School Contact, by clicking "Remove school contact" and following the process through.
- Once complete you should no longer see the School Contact.

## Link to Trello card

https://trello.com/c/tEvYmkM2/291-add-placement-contact-details-to-school-org-details

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated or [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

## Screenshots

https://github.com/DFE-Digital/itt-mentor-services/assets/17162488/e5f4ad6e-9e9c-485e-a87b-0e3fe63b3935

### CODE REVIEW

File: config/locales/en/shared/schools/location_details.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+en:
+  shared:
+    schools:
+      location_details:
+        address: Address
```
Comment (Nitemaeric) [inline config/locales/en/shared/schools/location_details.yml:5]: Let's add EOF newlines :)

File: config/locales/en/shared/schools/school_details.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+en:
+  shared:
+    schools:
+      school_details:
+        gias: Get Information about Schools (GIAS)
+        gias_details: "These details will be displayed to ITT providers. 
+          If any details are incorrect, go to the %{link_to} service to update them."
+        organisation_name: Organisation name
+        ukprn: UK provider reference number (UKPRN)
+        urn: Unique reference number (URN)
```
Comment (Nitemaeric) [inline config/locales/en/shared/schools/school_details.yml:10]: Here too :)

File: config/analytics_blocklist.yml [hunk 3]
```diff
@@ -1,64 +1,64 @@
...
-    - id
-    - key
-    - enabled
-    - created_at
-    - updated_at
+  - id
+  - key
+  - enabled
+  - created_at
+  - updated_at
```
Comment (Nitemaeric) [inline config/analytics_blocklist.yml:8]: I think the indentation here should be a level deeper, like it previously was.

File: config/analytics.yml [hunk 4]
```diff
@@ -161,3 +161,10 @@ shared:
     - created_at
     - updated_at
     - provider_id
+  :school_contacts:
+    - id
+    - name
+    - email_address
+    - school_id
+    - created_at
+    - updated_at
```
Comment (Nitemaeric) [inline config/analytics.yml:170]: EOF missing here.

File: config/analytics_blocklist.yml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 ---
-shared:
+:shared:
```
Comment (Nitemaeric) [inline config/analytics_blocklist.yml:2]: Is this change significant?

**Review by Nitemaeric — APPROVED** [review]
Just some minor bits pending. Claims side looks good! 👍

File: config/analytics_blocklist.yml (hunk 5 above)
Comment (JamieCleare2525) [reply config/analytics_blocklist.yml:2]: A lot of these changes were done automatically. The prefixed `:` seems to be consistent with the rest of the file though.

File: app/views/placements/schools/school_contacts/remove.html.erb
Comment (ollietreend) [inline app/views/placements/schools/school_contacts/remove.html.erb:1]: Is this view redundant? I don't think there's a link in the UI anywhere, nor an action in the controller.

Or am I being silly?

File: config/locales/en/placements/schools.yml [hunk 6]
```diff
@@ -8,3 +8,8 @@ en:
         ofsted: Ofsted
         send: Special educational needs and disabilities (SEND)
         contact_details: Contact details
```
Comment (ollietreend) [inline config/locales/en/placements/schools.yml:10]: Is this now redundant?

```suggestion
```

File: spec/models/placements/school_spec.rb [hunk 7]
```diff
@@ -69,6 +69,8 @@
 
 RSpec.describe Placements::School do
   context "with assocations" do
+    it { is_expected.to have_one(:school_contact) }
```
Comment (ollietreend) [inline spec/models/placements/school_spec.rb:72]: Ditto for the 'dependent' rule.

File: config/locales/en/shared/schools/school_contact_details.yml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+en:
+  shared:
+    schools:
+      school_contact_details:
+        no_placement_coordinator: This organisation does not have a school contact.
+        add_placement_coordinator: Add school contact
+        full_name: Full name
+        email: Email
+        change: Change
+        remove_placement_coordinator: Remove school contact
```
Comment (ollietreend) [inline config/locales/en/shared/schools/school_contact_details.yml:10]: We should probably rename these keys so they match the new name for this feature.

```suggestion
        no_school_contact: This organisation does not have a school contact.
        add_school_contact: Add school contact
        full_name: Full name
        email: Email
        change: Change
        remove_school_contact: Remove school contact
```

Comment (ollietreend) [inline config/locales/en/shared/schools/school_contact_details.yml:10]: Is this redundant?

```suggestion
```

**Review by ollietreend — COMMENTED** [review]
This is looking good 👍🏻 

Just a few things which I think are probably a result of the design changes.

File: app/views/placements/schools/school_contacts/remove.html.erb
Comment (JamieCleare2525) [reply app/views/placements/schools/school_contacts/remove.html.erb:1]: Ah yes. I removed the controller and tests, but forgot about the views.

File: app/helpers/placements/translations_helper.rb [hunk 9]
```diff
@@ -0,0 +1,6 @@
+module Placements::TranslationsHelper
+  def embedded_link_text(translations)
+    # translations.html_safe
```
Comment (ollietreend) [inline app/helpers/placements/translations_helper.rb:3]: Lingering comment 🙂 

```suggestion
```

**Review by ollietreend — APPROVED** [review]
Looks great! 🎉 

Just one lingering comment which you forgot to remove – but I'll approve anyway, if you don't mind removing it before merging 🙏🏻

**Pull request merged**